tests/extras/test_ftrl.py

```diff
@@ -298,32 +299,128 @@ def test_ftrl_set_wrong_inter_type():
 # Test getters and setters for wrong values of individual FTRL parameters
 #-------------------------------------------------------------------------------
 
-def test_ftrl_set_wrong_alpha_value():
+def test_ftrl_set_zero_alpha_value():
```

Comment:
You can simply parametrize this:
```python
@pytest.mark.parametrize('x', [0.0, math.nan, math.inf, None])
def test_ftrl_set_bad_alpha_value(x):
    ft = Ftrl()
    with pytest.raises(ValueError) as e:
        ft.alpha = x
    assert ("Value should be positive: %s" % str(x) == str(e.value))
```

Reply:
yes, forgot to re-factor it at the very end...